src/main/java/org/mockito/internal/creation/bytebuddy/InlineByteBuddyMockMaker.java

```diff
@@ -202,16 +207,25 @@ public InlineByteBuddyMockMaker() {
         }
         bytecodeGenerator =
                 new TypeCachingBytecodeGenerator(
-                        new InlineBytecodeGenerator(INSTRUMENTATION, mocks, mockedStatics), true);
+                        new InlineBytecodeGenerator(
+                                INSTRUMENTATION, mocks, mockedStatics, mockConstruction::get),
+                        true);
     }
 
     @Override
     public <T> T createMock(MockCreationSettings<T> settings, MockHandler handler) {
         Class<? extends T> type = createMockType(settings);
 
-        Instantiator instantiator = Plugins.getInstantiatorProvider().getInstantiator(settings);
         try {
-            T instance = instantiator.newInstance(type);
+            T instance;
+            try {
+                // We attempt to use the "native" mock maker first that avoids Objenesis and Unsafe
+                instance = newInstance(type);
```

Comment:
Why not _hiding_ this behind the interface ? The implementing class could still be this class.
And tweak the _plug-in selector_to use this one first. And use the fallback instantiator otherwise.